griptape/tools/email_client/tool.py

```diff
@@ -140,7 +141,13 @@ def retrieve(self, params: dict) -> ListArtifact | ErrorArtifact:
             Literal(
                 "body",
                 description="Email body"
-            ): str
+            ): str,
+            Optional(
+                Literal(
+                    "attachments",
+                    description="List of file paths to be attached to the email"
+                )
+            ): list[str]
```

Comment:
Do we keep this as a list paths? Would we want to load attachments from memory or should we recommend using this tool in conjunction with the `FileManager` tool that can store attachments in a separate action? Just something to think about for now.

Reply:
That's true, I just thought it's easier if this is done using one tool. And since it's just the EmailClient and GoogleGmailClient which we know needs attachments most of the time, we don't really need an extra tool (FileManager) to do this. Kind of easier for the user to have the functionality by calling the same tool.

Reply:
How would you, for example, attach a summary of a website to an email without `FileManager`?

Reply:
I would do that using the WebScraper tool to give a summary of a website and then send that as the content of the email. Why would you want that as an attachment though? Attaching a summary of a website to an email, as an actual attachment doesn't seem like a common use case of the attachment support according to me. If I was using `GoogleGmailClient` or `EmailClient`, I would simply want it to attach files for me from my local. I could be wrong though about the use case. You would know better.

Reply:
The website summary was just an example :)

The broader point is that this activity should be able to pull things from memory. If you have `FileManager` available, then the LLM can always load a file into memory and then send it as an attachment. And it can use any other tool to load up data before attaching it (including a website summary! :) ).

Reply:
Ok got it! Let me try to code it in a way we can use FileManager to do this.